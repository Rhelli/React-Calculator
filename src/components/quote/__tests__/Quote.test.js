import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Quote from '../Quote';

describe('The appearance of the Quote Page', () => {
  it('Matches the Quote Page snapshot', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Contains the Quote Page Outer Container', () => {
    const { container } = render(<Quote />);
    expect(container.firstChild.classList.contains('quotePage')).toBe(true);
  });

  it('Contains the Navbar', () => {
    const { container } = render(<Quote />);
    expect(container.firstChild.firstChild.classList.contains('navbarContainer')).toBe(true);
  });
})
