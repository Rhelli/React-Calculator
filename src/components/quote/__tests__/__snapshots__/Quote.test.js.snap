// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance of the Quote Page Matches the Quote Page snapshot 1`] = `
<div
  className="quotePage"
>
  <div
    className="navbarContainer"
  >
    <div
      className="navbarTitle"
    >
      <h1>
        Math Magicians
      </h1>
    </div>
    <div
      className="navbarLinks"
    >
      <a
        href="/"
      >
        Home 
      </a>
      <span>
         | 
      </span>
      <a
        href="/Calculator"
      >
        Calculator 
      </a>
      <span>
         | 
      </span>
      <a
        href="/Quote"
      >
        Quote 
      </a>
    </div>
  </div>
  <div
    className="quoteContainer"
  >
    <h3>
      "They don't think it be like it is, but it do"
    </h3>
    <h4>
      Oscar Gamble
    </h4>
  </div>
</div>
`;
